gateway/dataimpl/server_v1/collectionadminserver.go

```diff
@@ -158,6 +160,20 @@ func (s *CollectionAdminServer) CreateCollection(
 		return nil, s.errorHandler.NewGenericStatus(err).Err()
 	}
 
+	manifestUid, err := strconv.ParseUint(resp.ManifestUid, 16, 64)
+	if err != nil {
+		return nil, s.errorHandler.NewGenericStatus(err).Err()
+	}
+
+	opts := gocbcorex.EnsureManifestOptions{
+		BucketName:    in.BucketName,
+		CollectionUid: manifestUid,
+	}
+	err = bucketAgent.EnsureManifest(ctx, &opts)
+	if err != nil {
+		return nil, s.errorHandler.NewGenericStatus(err).Err()
```

Comment:
Is this definitely what we want to return or could we return something that could be handled by the user in some way? I'm not entirely sure how a caller would handle this error being returned at the moment. If they just retried the create then they'd get a different error (i.e. collection already exists) next time they call.

Reply:
Actually I think this is fine. It'll generally be a timeout.